source/client_channel_handler.c

```diff
@@ -835,15 +862,19 @@ static void s_mqtt_disconnect_task(struct aws_channel_task *channel_task, void *
     if (connection->state == AWS_MQTT_CLIENT_STATE_DISCONNECTING && connection->reconnect_task) {
         aws_atomic_store_ptr(&connection->reconnect_task->connection_ptr, NULL);
 
-        /* If the reconnect_task isn't scheduled, free it */
-        if (connection->reconnect_task && !connection->reconnect_task->task.timestamp) {
-            aws_mem_release(connection->reconnect_task->allocator, connection->reconnect_task);
+        /* Cancel if scheduled */
+        if (connection->reconnect_task->task.timestamp > 0 && connection->slot && connection->slot->channel) {
+            aws_event_loop_cancel_task(
+                aws_channel_get_event_loop(connection->slot->channel), &connection->reconnect_task->task);
         }
 
+        aws_mem_release(connection->reconnect_task->allocator, connection->reconnect_task);
```

Comment:
this won't work, you can't reliably cancel "channel tasks", they travel through too many queues (channel queue, event-loop queue, scheduler queue)
The previous approach that NULLed things out and let the task clean itself up was the better approach